Modules/pyexpat.c

```diff
@@ -1177,7 +1201,9 @@ xmlparse_dealloc(xmlparseobject *self)
         self->buffer = NULL;
     }
     Py_XDECREF(self->intern);
+    PyTypeObject *tp = Py_TYPE(self);
     PyObject_GC_Del(self);
```

Comment:
tp->tp_free(self) in here?

Reply:
@shihai1991 I haven't done that in other places where I switched to heap types.  Why is needed in this case?

Reply:
`tp_free` will hide the details: using `PyObject_GC_Del` or `PyObject_Del` will be the interned behavior of `tp_free`.
IMHO, it's more easy to understand. This just a suggestion ;)

Reply:
@shihai1991 I think I understand


for types which use Py_TPFLAGS_HAVE_GC these two are equivalent:
```
tp->tp_free(self);
PyObject_GC_Del(self);
```

and for types not using this flag these are equivalent:
```
tp->tp_free(self);
PyObject_Del(self);
```

So your suggestion is to always use `tp->tp_free(self)` so that the same patterns can be used whether the type is using GC or not.  Is that right?

Reply:
tp_free should be equal to PyObject_GC_Del, so I'm fine with calling PyObject_GC_Del directly. (The current code is fine.)